pkg/virt-controller/watch/workload-updater/workload-updater_test.go

```diff
@@ -164,7 +164,7 @@ var _ = Describe("Workload Updater", func() {
 
 			By("Checking prometheus metric before sync")
 			dto := &io_prometheus_client.Metric{}
-			outdatedVirtualMachineInstanceWorkloads.Write(dto)
+			Expect(outdatedVirtualMachineInstanceWorkloads.Write(dto)).To(Succeed())
```

Comment:
I wonder if it would make sense that the ginkgo linter from @nunnatsa would have an option to automatically assume on unchecked errors that this is the correct fix and that you can just auto-fix most of these and then just pick speical cases.

@nunnatsa what do you think?

Reply:
I'm not sure about it. We only want to do that for missing error checks within `It` functions (or table function etc.), or helper functions that are called from `It`. This is not so simple to recognize these cases. The linter skips files without gomega import, but still, this is not always the right thing to do even within a test file.

Reply:
I have no strong opinions, I would just recommend (for our project) to fix this mostly automatically in one go somehow (or initially ingore test files completely).

Reply:
I don't think ignoring test files is the way to go; I think some of the non-determinism of our test suites stems from  unhandled errors - its mostly a suspicion, I don't have proof currently.

... time may prove me wrong.

Reply:
> I don't think ignoring test files is the way to go; I think some of the non-determinism of our test suites stems from unhandled errors - its mostly a suspicion, I don't have proof currently.

I am not sure. the unit tests are mostly a little bit unstable where we test controllers  (virt-handler and virt-controller), and there we really have a too complex setup for tests, also involving go routines. From what I have seen checking for errors would not really improve the situation.

> ... time may prove me wrong.

Yeah, who knows. I just want to ensure that we invest as little time as possible. In this case it looks to me like 99% of all missing error checks should simply check if the command succeeds. You may be able to take advantage out of that.

Reply:
> I don't think ignoring test files is the way to go

This is not exactly what the linter does. It ignores files without gomega import. So it mostly focuses on test files, or test helper files, I guess.

Reply:
> > I don't think ignoring test files is the way to go
> 
> This is not exactly what the linter does. It ignores files without gomega import. So it mostly focuses on test files, or test helper files, I guess.

Can you elaborate on the context of your reply ? I'm very confused about this comment. I'm saying I think we shouldn't ignore errors unless there's a very good reason - hence, `errcheck` should - IMO - be used all over the place: production code / tests / ...